### PR TITLE
Store Promotions: Add product sale card

### PR DESCRIPTION
**Pull request by coderkevin**
This adds a card for a product sale promotion and the state editing to
go with it.

To Test:
1. `npm start`
2. Visit `http://calypso.localhost:3000/store/promotion/<site url>`
3. Select "Product Sale"
4. Click the radio button on several products.

(Note: Styling is not complete for this and will be completed in #18520)

new_promotion_ _coderkevin_ _wordpress_com

### CODE REVIEW

File: client/extensions/woocommerce/app/promotions/promotion-form-applies-to.js [hunk 1]
```diff
@@ -92,7 +94,17 @@ const renderProductCheckboxRow = ( appliesTo, promotion, onChange ) => ( product
...
+	const checked = isProductSelected( appliesTo, promotion, product.id );
+	return (
+		<div key={ product.id }>
+			<FormRadio value={ product.id } checked={ checked } onChange={ onChange } />
+			<FormLabel>{ product.name }</FormLabel>
+		</div>
+	);
+};
+
+const renderProductCategoryCheckboxRow = ( promotion, onChange ) => ( category ) => {
```
Comment (allendav) [inline client/extensions/woocommerce/app/promotions/promotion-form-applies-to.js]: Can't test this - I only get `Specific Product` in the `Applies to` dropdown

**Comment by allendav** [comment]
Casually reading this, I could see how a significant portion of users could confuse Amount with the discount amount (instead of the product Sale Price).  Maybe when Applies To is set to Specific Product we could have the "Amount" label say "Sale Price" instead?

screen shot 2017-10-06 at 11 42 06 am

**Comment by allendav** [comment]
When applies to is set to Specific Product, and I pick a product, I have no idea what the regular price of the product is. It would be nice to display that somewhere, possibly as a field explanation below the input box or something

screen shot 2017-10-06 at 11 48 35 am

**Comment by allendav** [comment]
Radio buttons seem to need some styling cc @kellychoffman @jameskoster

**Comment by justinshreve** [comment]
> Also worried about what happens when there are a bazillion products on a store.

I feel like ideally this would use something like https://github.com/Automattic/wp-calypso/tree/master/client/extensions/woocommerce/components/product-search instead of just spitting out a bunch of checkboxes.

**Comment by kellychoffman** [comment]
> Radio buttons seem to need some padding to separate them from the Applies To dropdown a bit

Thanks @allendav. @coderkevin - can you make sure to add us as reviewers to any PR with UI changes? And to add screenshots please.

**Comment by kellychoffman** [comment]
> When applies to is set to Specific Product, and I pick a product, I have no idea what the regular price of the product is. It would be nice to display that somewhere,

Can that be added directly to the product search component?

**Comment by coderkevin** [comment]
Rebased and adjusted for Jest/Prettier PRs.

**Comment by coderkevin** [comment]
The code in this PR has been implemented in other PRs at this point.

**Pull request closed without merging**